src/usb_stream.cpp

```diff
@@ -420,6 +420,10 @@ bool UsbStream::read_device(std::vector<uint8_t>& msg, const bool stream_on)
     else
     {
         return_value_ = read(file_id_, buffer_.data(), msg.size());
+        // try to read again if bytes requested != bytes received 
+        if (return_value_ != static_cast<ssize_t>(msg.size())){  // changed
+            return_value_ += read(file_id_, buffer_.data()+return_value_, msg.size()-return_value_);
```

Comment:
Can you remove the `changed`. Also, add spaces around the operands like `buffer_.data() + return_value_, msg.size() - return_value_`?

Reply:
Done.